unstructured/ingest/connector/astra.py

```diff
@@ -0,0 +1,116 @@
+import copy
+import uuid
+import typing as t
+
+from dataclasses import dataclass, field
+
+from unstructured.ingest.enhanced_dataclass import enhanced_field
+from unstructured.ingest.enhanced_dataclass.core import _asdict
+from unstructured.ingest.error import DestinationConnectionError, SourceConnectionNetworkError
+from unstructured.ingest.interfaces import (
+    AccessConfig,
+    BaseConnectorConfig,
+    BaseDestinationConnector,
+    WriteConfig,
+)
+from unstructured.ingest.logger import logger
+from unstructured.ingest.utils.data_prep import chunk_generator
+from unstructured.staging.base import flatten_dict
+from unstructured.utils import requires_dependencies
+
+
+if t.TYPE_CHECKING:
+    from astrapy.db import AstraDB, AstraDBCollection
+
+NON_INDEXED_FIELDS = ["metadata._node_content", "content"]
+
+
+@dataclass
+class AstraDBAccessConfig(AccessConfig):
+    token: t.Optional[str] = enhanced_field(default=None, sensitive=True)
+    api_endpoint: t.Optional[str] = enhanced_field(default=None, sensitive=True)
+
+
+@dataclass
+class SimpleAstraDBConfig(BaseConnectorConfig):
+    access_config: AstraDBAccessConfig
+    collection_name: str
+    embedding_dimension: int
+
+
+@dataclass
+class AstraDBWriteConfig(WriteConfig):
+    batch_size: int = 20
+
+
+@dataclass
+class AstraDBDestinationConnector(BaseDestinationConnector):
+    write_config: AstraDBWriteConfig
+    connector_config: SimpleAstraDBConfig
+    _astra_db = t.Optional["AstraDB"] = field(init=False, default=None)
```

Comment:
should be a : instead of =

Reply:
eek, bad typo! Thanks for catching that. Just updated it.